src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -330,6 +331,35 @@ def validate(self, node, options=None):
                 "transformation can only be applied to the outermost "
                 "Range.")
 
+        # If there is a loop variable defined in the config file and
+        # this variable is already defined in the code, is it defined
+        # as an integer?
+        array_index = node.position
+        loop_type_order = Config.get().api_conf("nemo").get_index_order()
+        loop_type_data = Config.get().api_conf("nemo").get_loop_type_data()
+        try:
+            loop_type = loop_type_order[array_index]
+            loop_type_info = loop_type_data[loop_type]
+            loop_variable_name = loop_type_info['var']
+            try:
+                symbol_table = node.scope.symbol_table
+                loop_variable_symbol = symbol_table.lookup(loop_variable_name)
+                # Check the existing loop variable name is a scalar integer
+                if not (loop_variable_symbol.is_scalar and
```

Comment:
That is possible and I modified the `validate` method to allow this even though I'm not sure that we want to support such a case. However, if so, when the transformation creates the explicit loop using `NemoLoop.create(...)`, this raises an exception as `create` expects the loop iterator to be a scalar integer. Therefore it is not possible to support this case, so instead I've modified the exception slightly to point out it might be a deferred type. Please let me know if you think there is a better option.